test/with_dummyserver/test_https.py

```diff
@@ -80,6 +90,27 @@ def test_verified(self):
                 error = call[0][1]
                 self.assertEqual(error, InsecurePlatformWarning)
 
+    def test_verified_context(self):
+        ctx = create_urllib3_context(cert_reqs=ssl.CERT_REQUIRED)
+        ctx.load_verify_locations(DEFAULT_CA)
+        https_pool = HTTPSConnectionPool(self.host, self.port,
+                                         ssl_context=ctx)
+
+        conn = https_pool._new_conn()
+        self.assertEqual(conn.__class__, VerifiedHTTPSConnection)
+
+        with mock.patch('warnings.warn') as warn:
```

Comment:
So this ended up being much more difficult than I expected. On top of `warnings.catch_warnings(record=True)` and `warnings.simplefilter("always")`, it is also necessary to reset __warningregistry__ in any modules you care about. Otherwise, depending on test order, any warnings with the "default" action will only be caught in the first test, and not any subsequent tests. I put together a context manager that extends `catch_warnings` to do this, see [here](https://github.com/divergentdave/urllib3/commit/e729ec2cfaea04204c4e164043eaf5ae2fa8a88b). How about we leave `mock.patch('warnings.warn')` for now, and look at the `catch_warnings` approach more generally in another PR?